### PR TITLE
LPD-1988: Allow to disassociate Response Schema from an API Endpoint through the UI

### PR DESCRIPTION
**Pull request by sergiojimcos**
## Motivation
The idea of this PR is to allow users to disassociate Response Schemas in API Endpoints (GET or POST) operations. However, at the moment according to Product it is not possible Request Schemas, in a future that validations will be deleted, so the solution is ready for the future.

**What is new?**
- Define new `unSelectedOption` in schemas options
- Add test cases and correct a flaky test.

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPD-1988

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/test/playwright/tests/headless-builder-web/post.spec.ts [hunk 1]
```diff
@@ -71,10 +71,49 @@ const studentSubjectsApplication = {
...
+
+	await page.getByRole('menuitem', {name: 'Not Selected'}).click();
+
+	await applicationPage.publishButton.click();
+
+	await expect(
+		page.getByText('Please select a request body schema.')
+	).toBeVisible();
+
+	await page.goto('/');
```
Comment (jaimelr10) [inline modules/test/playwright/tests/headless-builder-web/post.spec.ts:105]: Could we remove this?

File: modules/test/playwright/tests/headless-builder-web/post.spec.ts [hunk 2]
```diff
@@ -93,6 +132,12 @@ test('can create post endpoint and can not edit http method', async ({
 	);
 
 	await expect(isDisabled).toBeTruthy();
+
+	await page.goto('/');
```
Comment (jaimelr10) [inline modules/test/playwright/tests/headless-builder-web/post.spec.ts:136]: Could we also remove this?

File: modules/test/playwright/tests/headless-builder-web/get.spec.ts [hunk 3]
```diff
@@ -17,6 +17,78 @@ export const test = mergeTests(
...
+	await applicationPage.publishButton.click();
+
+	await page.waitForTimeout(2000);
+
+	await applicationPage.goToDetailsTab();
+	await applicationPage.goToEndpointsTab();
+	await applicationPage.goToEditEndpoint(endpoint.path);
+	await applicationPage.goToEndpointConfigurationTab();
+
+	await expect(page.getByLabel('Response Body Schema')).toBeVisible();
```
Comment (jaimelr10) [inline modules/test/playwright/tests/headless-builder-web/get.spec.ts:84]: This checks that there is not selected option? Or just that the dropdown exists. 
In that case, we should check that there is no selected schema, right?

**Review by jaimelr10 — CHANGES_REQUESTED** [review]
Maybe we should also rebase this with https://github.com/brianchandotcom/liferay-portal/pull/148068 to avoid conflicts

Right, @4lejandrito ?

File: modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/EditEndpointConfiguration.tsx [hunk 4]
```diff
@@ -53,15 +53,25 @@ export default function EditEndpointConfiguration({
 				: [];
 
 			if (options.length) {
-				setSchemaOptions(options);
+				const notSelectedOption = [
```
[the remark this answers is not in the document]
Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/EditEndpointConfiguration.tsx:56]: Nice!! Much better!! Thanks

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by sergiojimcos** [comment]
There are no related errors

**Comment by sergiojimcos** [comment]
Sent to Brian here: https://github.com/brianchandotcom/liferay-portal/pull/148308

**Pull request closed without merging**